File: app/src/main/java/com/example/ikastyle/Dao/MainCategoryDao.java
package com.example.ikastyle.Dao;

import androidx.room.Dao;
import androidx.room.Query;

import com.example.ikastyle.Entity.MainCategory;

import java.util.List;

@Dao
public interface MainCategoryDao {
    @Query("SELECT * FROM MAST_MAIN_CATEGORY WHERE language_code = :languageCode ORDER BY id")
    List<MainCategory> getMainCategoryList(int languageCode);
}

File: app/src/main/java/com/splatool/ikastyle/model/data/dao/MainCategoryDao.kt
package com.splatool.ikastyle.model.data.dao

import com.splatool.ikastyle.model.data.entity.MainCategory
import androidx.room.*

@Dao
interface MainCategoryDao {
    @Query("SELECT * FROM MAST_MAIN_CATEGORY WHERE language_code = :languageCode ORDER BY id")
    suspend fun getMainCategoryList(languageCode: Int): List<MainCategory>


    @Query("SELECT * FROM MAST_MAIN_CATEGORY WHERE language_code = :languageCode ORDER BY id")
    fun getMainCategoryList_nonSuspend(languageCode: Int): List<MainCategory>
}